File: assignment/269p3-5.R
pima <- read.csv('./data/pima2.csv', header = T)

classify <- function(pregnant) {
    if (pregnant <= 5) {
        return('0~5')
    }else if (6 <= pregnant && pregnant <= 10) {
        return('6~10')
    }else {
        return('10+')
    }
}


for (col in c('glucose', 'pressure', 'triceps', 'insulin', 'mass', 'pedigree')) {
    print(paste0('**********:', col))
    print('##########:MEAN')
    print(aggregate(pima[col], list(diabetes = pima$diabetes), mean, na.rm = T))
    print('##########:SD')
    print(aggregate(pima[col], list(diabetes = pima$diabetes), sd, na.rm = T))
}
pima$pregnant <- unlist(lapply(pima$pregnant, classify))

for (col in c('glucose', 'pressure', 'triceps', 'insulin', 'mass', 'pedigree')) {
    print(paste0('**********:', col))
    print('##########:MEAN')
    print(aggregate(pima[col], list(diabetes = pima$pregnant), mean, na.rm = T))
    print('##########:SD')
    print(aggregate(pima[col], list(diabetes = pima$pregnant), sd, na.rm = T))
}

File: assignment/269p3-5.py
import pandas as pd

pima = pd.read_csv('../data/pima2.csv')
for col in ['glucose', 'pressure', 'triceps', 'insulin', 'mass', 'pedigree']:
    print('#' * 10 + 'diabetes:' + col + ':mean')
    data = pima.groupby('diabetes').describe()[col]
    print(data['mean'])
    print('#' * 10 + 'diabetes:' + col + ':std')
    print(data['std'])


def classify(pregnant):
    if pregnant <= 5:
        return '0~5'
    elif 6 <= pregnant <= 10:
        return '6~10'
    else:
        return '10+'


pima['pregnant'] = pima['pregnant'].apply(classify)
for col in ['glucose', 'pressure', 'triceps', 'insulin', 'mass', 'pedigree']:
    print('#' * 10 + 'pregnant:' + col+':mean')
    data = pima.groupby('pregnant').describe()[col]
    print(data['mean'])
    print('#' * 10 + 'pregnant:' + col+':std')
    print(data['std'])
